File: AGCV2/AGC4WApi/ClassEx/SystemSet/clsCacheModeExWApi.cs
﻿
 /*-- -- -- -- -- -- -- -- -- -- --
 类名:clsCacheModeExWApi
 表名:CacheMode(00050538)
 生成代码版本:2020.05.28.1
 生成日期:2020/05/28 17:10:36
 生成者:
 生成服务器IP:192.168.1.10
 工程名称:AGC
 工程ID:0005
 相关数据库:tzar.tpddns.cn,19433AGC_CS12
 PrjDataBaseId:0213
 模块中文名:字段、表维护
 模块英文名:Table_Field
 框架-层名:WA_访问扩展层(WA_AccessEx)
 编程语言:CSharp
 注意:1、需要数据底层(PubDataBase.dll)的版本:2019.03.07.01
        2、需要公共函数层(TzPubFunction.dll)的版本:2017.12.21.01
 == == == == == == == == == == == == 
 */
using System;
using System.Data; 
using System.Data.SqlClient;
using System.Text; 
using System.Web;
using System.Collections; 
using com.taishsoft.common;
using System.ServiceModel;
using AGC.Entity;
using System.Collections.Generic;
using com.taishsoft.json;
using Newtonsoft.Json.Linq; 

namespace AGC4WApi
{
public static class clsCacheModeExWApi_Static
{

 /// <summary>
 /// 把同一个类的对象,复制到另一个对象
 /// (AutoGCLib.WA_AccessEx4CSharp:Gen_4WAEx_Static_CopyToEx)
 /// </summary>
 /// <param name = "objCacheModeENS">源对象</param>
 /// <returns>目标对象=>clsCacheModeEN:objCacheModeENT</returns>
 public static clsCacheModeENEx CopyToEx(this clsCacheModeEN objCacheModeENS)
{
try
{
 clsCacheModeENEx objCacheModeENT = new clsCacheModeENEx();
clsCacheModeWApi.CopyTo(objCacheModeENS, objCacheModeENT);
 return objCacheModeENT;
}
catch (Exception objException)
{
string strMsg = string.Format("(errid:Watl000002)Copy表对象数据出错,{1}.({0})",
clsStackTrace.GetCurrClassFunction(),
objException.Message); 
throw new Exception(strMsg); 
}
}

 /// <summary>
 /// /// 功能:通过SQL命令来修改记录,该方式是优化方式
 /// /// 优点:1、能够处理字段中的单撇问题；2、能够处理脏字段,即只有修改过的字段才需要修改同步到数据库.
 /// (AutoGCLib.WA_AccessEx4CSharp:Gen_4WAEx_Static_EditRecordEx)
 /// </summary>
 /// <param name = "objCacheModeEN">需要添加的实体对象</param>
 /// <returns>如果修改成功则返回TRUE,否则为FALSE</returns>
public static bool EditRecordEx(this clsCacheModeEN objCacheModeEN)
{
try
{
objCacheModeEN.sfUpdFldSetStr = objCacheModeEN.getsfUpdFldSetStr();
bool bolResult = clsCacheModeExWApi.EditRecordEx(objCacheModeEN);
// 静态的对象列表,用于清空相关缓存,针对记录较少,作为参数表可以使用
clsCacheModeWApi.ReFreshCache();
return bolResult;
}
catch (Exception objException)
{
string strMsg = string.Format("(errid:Watl000006)编辑记录出错,{1}!(from {0})",
clsStackTrace.GetCurrClassFunction(),
objException.Message); 
throw new Exception(strMsg); 
}
}
}
 /// <summary>
 /// 缓存方式(CacheMode)
 /// (AutoGCLib.WA_AccessEx4CSharp:GeneCode)
 /// </summary>
public class clsCacheModeExWApi
{
private static readonly string mstrApiControllerName = "CacheModeExApi";
 /// <summary>
 /// 静态的对象列表,用于缓存,针对记录较少,作为参数表可以使用
 /// (AutoGCLib.WA_AccessEx4CSharp:GeneCode)
 /// </summary>
public static List<clsCacheModeEN> arrCacheModeObjLst_Cache = null;
 /// <summary>
 /// 从缓存中查找失败的次数
 /// (AutoGCLib.WA_AccessEx4CSharp:GeneCode)
 /// </summary>
protected static int intFindFailCount = 0;

/// <summary>
/// 构造函数
 /// (AutoGCLib.WA_AccessEx4CSharp:Gen_4WAEx_ClassConstructor1)
/// </summary>
 public clsCacheModeExWApi()
 {
 }

 /// <summary>
 /// 修改记录
 /// (AutoGCLib.WA_AccessEx4CSharp:Gen_4WAEx_EditRecordEx)
 /// </summary>
 /// <returns>是否成功?</returns>
public static bool EditRecordEx(clsCacheModeEN objCacheModeEN)
{
if (string.IsNullOrEmpty(objCacheModeEN.sfUpdFldSetStr) == true)
{
string strMsg = string.Format("修改时，修改标志串为空，请联系管理员.对象关键字:{0}.(from {1}).",
objCacheModeEN.CacheModeId, clsStackTrace.GetCurrClassFunction());
throw new Exception(strMsg);
}
string strAction = "EditRecordEx";
string strErrMsg = string.Empty;
string strResult = "";
Dictionary<string, string> dictParam = new Dictionary<string, string>();
try
{
string strJson = clsJSON.GetJsonFromObj<clsCacheModeEN>(objCacheModeEN);
if (clsPubFun4WApi.Post(mstrApiControllerName, strAction, dictParam, strJson, out strResult, out strErrMsg) == true)
{
JObject jobjReturn = JObject.Parse(strResult);
if ((int)jobjReturn["errorId"] == 0)
{
var bolReturnBool = (bool)jobjReturn["returnBool"];
return bolReturnBool;
}
else
{
string strMsg = string.Format("{0}", jobjReturn["errorMsg"]);
throw new Exception(strMsg);
}
}
else return false;
}
catch (Exception objException)
{
 string strMsg = string.Format("执行WebApi功能出错, {0}.(from {1}). WebApi地址:{2}).",
      HttpUtility.UrlDecode(objException.Message), clsStackTrace.GetCurrClassFunction(),
clsPubFun4WApi.GetWebApiUrl(mstrApiControllerName, strAction));
 throw new Exception(strMsg);
}
}

}
}

File: GraduateStudyWebApp/GSSpa/Pages/TS/L3_ForWApiEx/SystemSet/clsCacheModeExWApi.ts
﻿
 /**
 * 类名:clsCacheModeExWApi
 * 表名:CacheMode(01120688)
 * 生成代码版本:2021.09.17.1
 * 生成日期:2021/09/20 16:03:13
 * 生成者:pyf
 * 生成服务器IP:103.116.76.183
 * 工程名称:问卷调查
 * 工程ID:0112
 * 相关数据库:103.116.76.183,9433EduHigh_Jsie
 * PrjDataBaseId:0170
 * 模块中文名:系统设置
 * 模块英文名:SystemSet
 * 框架-层名:WA_访问扩展层(WA_AccessEx)
 * 编程语言:TypeScript
 * 注意:1、需要数据底层(PubDataBase.dll)的版本:2019.03.07.01
   *      2、需要公共函数层(TzPubFunction.dll)的版本:2017.12.21.01
 **/

 /**
 * 缓存方式(CacheMode)
 * (AutoGCLib.WA_AccessEx4TypeScript:GeneCode)
* Created by pyf on 2021年09月20日.
* 注意:该类必须与调用界面处于同一个包，否则调用不成功!
 **/
//import $ from "jquery";
import { clsCacheModeEN } from "../../L0_Entity/SystemSet/clsCacheModeEN.js";
import { clsCacheModeENEx } from "../../L0_Entity/SystemSet/clsCacheModeENEx.js";
import { clsSysPara4WebApi2 } from "../../PubFun/clsSysPara4WebApi2.js";
import { clsString } from "../../PubFun/clsString.js";
import { CacheMode_CopyObjTo } from "../../L3_ForWApi/SystemSet/clsCacheModeWApi.js";
export class clsCacheModeExWApi 
{
 public static mstrController = "CacheModeExApi";
public objCacheModeEN = new clsCacheModeEN();

 /**
 * 获取WebApi的地址
 * (AutoGCLib.WA_AccessEx4TypeScript:Gen_4WAEx_Ts_GetWebApiUrl)
 * @returns 返回当前文件中Web服务的地址
 **/
public static GetWebApiUrl(strController: string, strAction: string): string {
let strServiceUrl:string;
if (clsSysPara4WebApi2.bolIsLocalHost == false) {
strServiceUrl = clsString.Format("{0}/{1}/{2}/{3}", clsSysPara4WebApi2.CurrIPAddressAndPort, clsSysPara4WebApi2.CurrPrx, strController, strAction);
}
else {
strServiceUrl = clsString.Format("{0}/{1}/{2}/{3}", clsSysPara4WebApi2.CurrIPAddressAndPort_Local, clsSysPara4WebApi2.CurrPrx, strController, strAction);
}
return strServiceUrl;
}

 /**
 * 把同一个类的对象,复制到另一个对象
 * (AutoGCLib.WA_AccessEx4TypeScript:Gen_4WAEx_Ts_CopyToEx)
 * @param objCacheModeENS:源对象
 * @returns 目标对象=>clsCacheModeEN:objCacheModeENT
 **/
 public static CopyToEx(objCacheModeENS:clsCacheModeEN ): clsCacheModeENEx
{
 const objCacheModeENT = new clsCacheModeENEx();
try
{
CacheMode_CopyObjTo(objCacheModeENS, objCacheModeENT);
 return objCacheModeENT;
}
catch (e)
{
const strMsg = clsString.Format("(errid:Watl000066)Copy表对象数据出错,{0}.", e);
console.error(strMsg);
alert(strMsg);
 return objCacheModeENT;
}
}

}